lib/spack/spack/schema/mirrors.py

```diff
@@ -17,7 +17,19 @@
         'default': {},
         'additionalProperties': False,
         'patternProperties': {
-            r'\w[\w-]*': {'type': 'string'},
+            r'\w[\w-]*': {
```

Comment:
Mirrors can now have separate `push` and `fetch` URLs (like git), for the case where a mirror may be accessed via a mechanism that differs from that with which resources are pushed to it, such as a mirror accessed through a web server, but populated by copying files to a path on the local file system.